pyroll/core/roll_pass/hookimpls/roll.py

```diff
@@ -25,7 +26,11 @@ def contact_length_square_oval(self: RollPass.Roll):
 
 @RollPass.Roll.contact_area
 def contact_area(self: RollPass.Roll):
-    return (self.roll_pass.in_profile.width + self.roll_pass.out_profile.width) / 2 * self.contact_length
+    if "3fold" in self.roll_pass.classifiers:
+        in_profile_local_width = self.roll_pass.in_profile.local_width(-self.roll_pass.in_profile.height / 2 * 0.999)
```

Comment:
I think the problem is occurrence of empty intersection line in `local_width`. This should not occur for squares for example with width-parallel boundary. Please capture this problem there and avoid such numerical factors.